format/udf-spec.md

```diff
@@ -0,0 +1,285 @@
+| *optional*  | `location`        | `string`               | Storage location of metadata files.                                                                             |
+| *optional*  | `properties`      | `map`                  | A string to string map of properties.                                                                           |
+| *optional*  | `secure`          | `boolean`              | Whether it is a secure function. Default: `false`.                                                              |
+| *optional*  | `doc`             | `string`               | Documentation string.                                                                                           |
+
+Notes:
+1. When `secure` is `true`,
+    - Engines **SHOULD NOT** expose the function definition through any inspection (e.g., `SHOW FUNCTIONS`).
+    - Engines **SHOULD** ensure that execution does not leak sensitive information through any channels, such as error messages, logs, or query plans.
+   
+### Overload
+
+Function overloads allow multiple implementations of the same function name with different signatures. Each overload has
+the following fields:
+
+| Requirement | Field name                 | Type                                                   | Description                                                                                   |
+|-------------|----------------------------|--------------------------------------------------------|-----------------------------------------------------------------------------------------------|
+| *required*  | `overload-id`              | `long`                                                 | Monotonically increasing identifier of this function overload.                                |
+| *required*  | `parameters`               | `list<parameter>`                                      | Ordered list of [function parameters](#parameter). Invocation order **must** match this list. |
+| *required*  | `return-type`              | `string` (Iceberg data type)                           | Return type. Example: `"string"`, `"struct<...>"`.                                            |
+| *required*  | `versions`                 | `list<overload-version>`                               | [Versioned implementations](#overload-version) of this overload.                              |
+| *required*  | `current-overload-version` | `long`                                                 | Monotonically increasing identifier of the current overload version.                          |
+| *optional*  | `function-type`            | `string` (either `"udf"` or `"udtf"`, default `"udf"`) | If `"udtf"`, `return-type` must be a `struct` describing the output schema.                   |
+| *optional*  | `doc`                      | `string`                                               | Documentation string.                                                                         |
+
+### Parameter
```

Comment:
Do we want to mention that the parameters do not have default values?

Reply:
Added a note to call it out that parameters have no default value. Thanks for the suggestions. We may think about support it later. Snowflake[1] and Postgres support it. Spark[2] plans to support it. 

1. “Each optional argument has a default value that is used when the argument is omitted.” –[ Snowflake Docs](https://docs.snowflake.com/en/developer-guide/udf/udf-calling-sql) ·[ Snowflake Blog](https://medium.com/snowflake/snowflake-supports-named-and-optional-arguments-389d2500726f)
2. [UDFs with default values for parameters as a future improvement](https://www.databricks.com/blog/2021/10/20/introducing-sql-user-defined-functions.html)

Reply:
> define the cases where you must use the same set of parameters. 

This was done when we uses the type tuples as definition id. It guarantees that functions with the same type tuples cannot co-exist, even their return types are different.